### ISSUE TITLE
Bump k8s 1.23.3 to get bug fixes

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kubebuilder#2502]
### What do you want to happen?

Bump k8s 1.23.3 to get bug fixes

**Comment by camilamacedo86** [comment]
We need also to bump controller-runtime before.
Then, bump the new c+r version on kb and the k8s version used in the Kubebuilder scaffolds. 

c/c @varshaprasad96

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 
- I am understanding about bumping `sigs.k8s.io/controller-runtime` version, but what do you mean by 

> Bump k8s 1.23.3 to get bug fixes

Because k8s alone is not a package in `go.mod`. Could you please elaborate.

**Comment by camilamacedo86** [comment]
Here we need to:
- Bump all k8s deps first in controller-runtime: https://github.com/kubernetes-sigs/controller-runtime/blob/master/go.mod#L19-L23 (I think now can be 0.23.4)
- Then, align with them to get a new release (we can ask in the channels for help)
- Bump controller runtime in Kubebulder project + all related deps here to match with the same k8s ones used. 
e.g.; https://github.com/kubernetes-sigs/kubebuilder/pull/2340

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 Update k8s dependencies in kubernetes-sigs/controller-runtime. https://github.com/kubernetes-sigs/controller-runtime/pull/1843

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 Added a comment on slack https://kubernetes.slack.com/archives/CAR30FCJZ/p1647978366266989
regarding the new release of kubernetes-sigs/controller-runtime. We can start working on this issue after the new release, right ?

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 Added a comment on slack https://kubernetes.slack.com/archives/CAR30FCJZ/p1647978366266989
regarding the new release of kubernetes-sigs/controller-runtime. We can start working on this issue only after the new release, right ?

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 Added a comment on slack https://kubernetes.slack.com/archives/CAR30FCJZ/p1647978366266989
regarding the new release of kubernetes-sigs/controller-runtime few days ago. We can start working on this issue only after the new release, right ?

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 Made another PR upgrading k8s dependencies for branch `release-0.11` in https://github.com/kubernetes-sigs/controller-runtime. Since it is merged now, can we now start working on this issue ?

**Comment by camilamacedo86** [comment]
To close this one is missing bump C+r and then ensure that all scaffolds will also be updated.

**Comment by NikhilSharmaWe** [comment]
@camilamacedo86 
Could you please confirm, So we have to - 
- update controller runtime version here https://github.com/NikhilSharmaWe/kubebuilder/blob/bumpK8s/pkg/plugins/golang/v3/scaffolds/init.go#L33.
- update `ENVTEST_K8S_VERSION` in https://github.com/NikhilSharmaWe/kubebuilder/blob/bumpK8s/pkg/plugins/golang/v3/commons.go#L81.
- update k8s version in https://github.com/NikhilSharmaWe/kubebuilder/blob/bumpK8s/test/common.sh#L28 and https://github.com/NikhilSharmaWe/kubebuilder/blob/bumpK8s/test/common.sh#L48.
And run make generate.

Could you please inform whether this is what needs to be done or not.
